controllers/datadogagent/feature/asm/feature.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package asm
+
+import (
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/merger"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	err := feature.Register(feature.ASMIDType, buildASMFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildASMFeature(options *feature.Options) feature.Feature {
+	asmFeat := &asmFeature{}
+
+	return asmFeat
+}
+
+type asmFeature struct {
+	owner                    metav1.Object
+	clusterAgentEnvAdditions []*corev1.EnvVar
+}
+
+// ID returns the ID of the Feature
+func (f *asmFeature) ID() feature.IDType {
+	return feature.ASMIDType
+}
+
+func (f *asmFeature) shouldEnableASM(asm *v2alpha1.ASMFeatureConfig) bool {
+	return apiutils.BoolValue(asm.SCA.Enabled) || apiutils.BoolValue(asm.Threats.Enabled) || apiutils.BoolValue(asm.IAST.Enabled)
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *asmFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
+	f.owner = dda
+	asm := dda.Spec.Features.ASM
+	if !f.shouldEnableASM(asm) {
+		return feature.RequiredComponents{}
+	}
+
+	if apiutils.BoolValue(asm.Threats.Enabled) {
+		f.clusterAgentEnvAdditions = append(f.clusterAgentEnvAdditions, &corev1.EnvVar{
+			Name:  apicommon.DDAdmissionControllerAppsecEnabled,
+			Value: "true",
+		})
+	}
+
+	if apiutils.BoolValue(asm.IAST.Enabled) {
+		f.clusterAgentEnvAdditions = append(f.clusterAgentEnvAdditions, &corev1.EnvVar{
+			Name:  apicommon.DDAdmissionControllerIASTEnabled,
+			Value: "true",
+		})
+	}
+
+	if apiutils.BoolValue(asm.SCA.Enabled) {
+		f.clusterAgentEnvAdditions = append(f.clusterAgentEnvAdditions, &corev1.EnvVar{
+			Name:  apicommon.DDAdmissionControllerAppsecSCAEnabled,
+			Value: "true",
+		})
+	}
+
+	// The cluster agent and the admission controller are required for the ASM feature.
```

Comment:
should we add a check in `shouldEnableASM` for the admission controller being enabled?

perhaps we can also add a warn log if it isn't

Reply:
Correct me if I am wrong but I don't think we have access to a logger in this place

Reply:
Yes logging isn't available right now. However still a question whether it make sense to add these configs when admission controller isn't enabled? 
Sample how it's done in other case 
https://github.com/DataDog/datadog-operator/blob/3e962f4c0b85585313bebda77fc39e1a557e0a56/controllers/datadogagent/feature/apm/feature.go#L128-L129

Reply:
Scratch that ^^, you added it already.